driver-core/src/main/com/mongodb/internal/TimeoutContext.java

```diff
@@ -174,12 +174,24 @@ public void resetTimeout() {
         timeout = calculateTimeout(timeoutSettings.getTimeoutMS());
     }
 
+    public TimeoutContext withAdditionalReadTimeout(final int additionalReadTimeout) {
+        long newReadTimeout = getReadTimeoutMS() + additionalReadTimeout;
+        newReadTimeout = newReadTimeout > 0 ? newReadTimeout : Long.MAX_VALUE;
+        if (timeout != null) {
+            if (timeout.isInfinite()) {
+                return this;
+            }
+            return new TimeoutContext(timeoutSettings.withTimeoutMS(newReadTimeout));
```

Comment:
Could you help me understand the purpose of the `timeout != null` check in this method? It seems that timeout is always set to null in the n`onUserOperationContext` method.

Also, I'm curious about the reasoning behind assigning `newReadTimeout` to `timeoutMs`. Is it intended for a future use case?

Reply:
Good catch - have added an assertion to ensure there is no `timeoutMS` set.